cyber_source/download_payment_batch_detail_report.rb

```diff
@@ -113,6 +114,31 @@ def main
     puts e.backtrace
   end
 
+  def credit_owner(payload)
+    location = payload[:location]
+    new_payload = {}
+    fine_owner = ''
+
+    if location.nil?
+      fine_owner = payload[:library]
+    elsif location =~ /^Lane/
+      fine_owner = 'Lane'
+    elsif location =~ /^Business/
+      fine_owner = 'Business'
+    else
```

Comment:
If I understand correctly, with this solution if a location in say Green Library has the word "Lane" (as in Lane Reading Room) or "Business" then it will be assigned the wrong owner.  To ward against that (and having to remember not to put "Lane" or "Business" as part of a location name... and Spec has a "Spec Law" location) would it make sense to have a library/location combo test?  location =~ /^Lane/ && library =~ /^SAL3/   ?? (not sure if && is correct for "and" here)

Reply:
Here the word Lane or Business would have to be at the beginning of the location name (not just anywhere in the location name). The only issue with checking the library is that it is not part of the [accounts json model](https://s3.amazonaws.com/foliodocs/api/mod-feesfines/p/accounts.html#accounts_get). I'd have to make one (maybe two) additional api calls to find that.

Reply:
I think the "Green Lane Reading Room" locations are actually "Green HAS {something}" as the folio location name. "Green Lane Reading Room" is the `discoveryDisplayName` See: https://drive.google.com/drive/folders/1EyaO2EInNKHFq8Kx0q-KbYPGGqAnGSOs

Reply:
True, since you're checking the beginning of the location name, we're good.  I didn't compute that (even as I typed the ^ for my example, doh)

Reply:
I mean this is still not really the most ideal solution since we need to remember not to create new locations that may fall into that trap. But it may be unlikely at this point that we would do that.

Reply:
The standard we've agreed on for location names has the library as first word, so this should be fine. (If someone deviates from the standard it will cause issues in multiple places probably)